### PR TITLE
Use dfmt on std/concurrency.d

### PR DESCRIPTION
**Pull request by JackStouffer**
std/concurrency.d uses a non standard style, this brings it into line with the rest of the codebase

used the default dfmt options with some manual fixes.

CC @wilzbach

### CODE REVIEW

**Comment by Hackerpilot** [comment]
Small request: also search for `" ~ "` in the formatted result and delete it.

**Comment by JackStouffer** [comment]
@Hackerpilot Done

File: std/concurrency.d [hunk 1]
```diff
@@ -1347,59 +1262,56 @@ class FiberScheduler :
...
         {
             super(m);
             notified = false;
         }
 
         override void wait() nothrow
         {
-            scope(exit) notified = false;
+            scope (exit)
+                notified = false;
```
Comment (wilzbach) [inline std/concurrency.d]: I am not sure whether this is defined by the style guide

File: std/concurrency.d [hunk 2]
```diff
@@ -1504,7 +1415,9 @@ private:
     auto cond = fs.newCondition(mtx);
 
     size_t received, sent;
-    auto waiter = new Fiber({receive(cond, received);}), notifier = new Fiber({send(cond, sent);});
+    auto waiter = new Fiber({ receive(cond, received); }), notifier = new Fiber({
+        send(cond, sent);
+    });
     waiter.call();
```
Comment (wilzbach) [inline std/concurrency.d]: This looks a bit weird as newlines are used only for the second `Fiber` expression

File: std/concurrency.d [hunk 3]
```diff
@@ -1751,8 +1648,7 @@ void yield(T)(T value)
                 assert(i == 4);
             });
 
-            auto r = new Generator!int(
-            {
+            auto r = new Generator!int({
```
Comment (wilzbach) [inline std/concurrency.d]: Again this isn't defined by the style guide

File: std/concurrency.d (hunk 1 above)
Comment (JackStouffer) [inline std/concurrency.d]: Will fix this

File: std/concurrency.d (hunk 2 above)
Comment (JackStouffer) [inline std/concurrency.d]: ditto

File: std/concurrency.d (hunk 3 above)
Comment (JackStouffer) [inline std/concurrency.d]: It's consistent with then ending `});`, so I'm for it

**Comment by JackStouffer** [comment]
@wilzbach Fixed

File: std/concurrency.d [hunk 4]
```diff
@@ -781,32 +747,19 @@ body
...
-        ( Variant val )
-        {
-            static if (T.length > 1)
-                string exp = T.stringof;
-            else
-                string exp = T[0].stringof;
+    thisInfo.ident.mbox.get((T val) {
+        static if (T.length)
+            ret.field = val;
+    }, (LinkTerminated e) { throw e; }, (OwnerTerminated e) { throw e; }, (Variant val) {
```
Comment (wilzbach) [inline std/concurrency.d]: I would use one for each statement, e.g.

```
(LinkTerminated e) { throw e; },
(OwnerTerminated e) { throw e; },
```

but it's already a huge improvement of the status quo!

File: std/concurrency.d [hunk 5]
```diff
@@ -842,80 +795,69 @@ body
...
+    static assert(!__traits(compiles, {
+        receiveTimeout(msecs(0), (int x) {  }, (int x) {  });
+    }));
 
-    static assert( __traits( compiles,
-                      {
-                          receiveTimeout( msecs(10), (int x) {}, (Variant x) {} );
-                      } ) );
+    static assert(__traits(compiles, {
+        receiveTimeout(msecs(10), (int x) {  }, (Variant x) {  });
```
Comment (wilzbach) [inline std/concurrency.d]: Is there any reason why you convert `{}` to `{ }`?
AFIAK `{}` is a very common style in Phobos (485 occurrences).

File: std/concurrency.d [hunk 6]
```diff
@@ -1013,21 +947,20 @@ private void unregisterMe()
  *  true if the name is available and tid is not known to represent a
  *  defunct thread.
  */
-bool register( string name, Tid tid )
+bool register(string name, Tid tid)
 {
-    synchronized( registryLock )
+    synchronized (registryLock)
```
Comment (wilzbach) [inline std/concurrency.d]: Q: Should we enforce a whitespace after `synchronized`  within the entire Phobos codebase by adding this to the checks to the `posix.mak`?

File: std/concurrency.d [hunk 7]
```diff
@@ -1422,57 +1333,56 @@ private:
         void switchContext() nothrow
         {
             mutex_nothrow.unlock_nothrow();
-            scope(exit) mutex_nothrow.lock_nothrow();
+            scope (exit) mutex_nothrow.lock_nothrow();
```
Comment (wilzbach) [inline std/concurrency.d]: There are 271 occurrences of `scope(exit)` in Phobos, so maybe we should agree on a common style here as well?

**Comment by wilzbach** [comment]
(the Dscanner failure is due to the random behavior of `std.parallelism` and the new Project tester seems to have some configuration issues for now.)

File: std/concurrency.d (hunk 6 above)
Comment (JackStouffer) [inline std/concurrency.d]: yes

File: std/concurrency.d (hunk 7 above)
Comment (JackStouffer) [inline std/concurrency.d]: Same line if one liner. Apply other rules for block statements to `scope` when there are multiple lines.

**Pull request merged**